sympy/simplify/cse_main.py

```diff
@@ -40,6 +40,197 @@
 # ===============================================================
 
 
+class CseExpr(tuple):
+    """Type of objects to be returned by the ``cse`` function.
+
+    ``CseExpr`` is backwards compatible with previous objects returned by the
+    ``cse`` function, which took the form:
+
+    ``Tuple[List[Tuple[Symbol, Expr]], List[Expr]]``
+
+    Examples
+    ========
+
+    A ``CseExpr`` can be constructed from a ``tuple`` containing two items, the
+    first being a ``list`` of symbol-expression substitution pairs, and the
+    second being a ``list`` of reduced expressions that contain the substitution
+    symbols:
+
+    >>> from sympy import symbols, sin, exp
+    >>> from sympy.simplify.cse_main import CseExpr
+    >>> x0, x1, x2, x3 = symbols('x0, x1, x2, x3')
+    >>> cse_expr = CseExpr(([(x0, x1 / x2), (x3, x0 - exp(x2))],
+    ...                     [x3 * (x3 + sin(x0))]))
+    >>> print(cse_expr)
+    ([(x0, x1/x2), (x3, x0 - exp(x2))], [x3*(x3 + sin(x0))])
+
+    The ``list`` of symbol-expression substitution pairs can be accessed using
+    either the ``subs`` attribute or using the zeroth index:
+
+    >>> print(cse_expr.subs)
+    [(x0, x1/x2), (x3, x0 - exp(x2))]
+    >>> print(cse_expr[0])
+    [(x0, x1/x2), (x3, x0 - exp(x2))]
+
+    The ``list`` of substituted expressions can also be accessed using either
+    the ``exprs`` attribute or using the first index:
+
+    >>> print(cse_expr.exprs)
+    [x3*(x3 + sin(x0))]
+    >>> print(cse_expr[1])
+    [x3*(x3 + sin(x0))]
+
+    The symbol-expression substitution pairs can also be accessed as a ``dict``
+    mapping symbols as keys to expressions as values:
+
+    >>> print(cse_expr.subs_mapping)
+    {x0: x1/x2, x3: x0 - exp(x2)}
+
+    If a pure ``tuple`` is required for backwards compatibility, the
+    ``as_tuple`` method can be used to generate this:
+
+    >>> cse_expr_as_tuple = cse_expr.as_tuple()
+    >>> print(cse_expr_as_tuple)
+    ([(x0, x1/x2), (x3, x0 - exp(x2))], [x3*(x3 + sin(x0))])
+    >>> print(isinstance(cse_expr_as_tuple, tuple))
+    True
+    >>> print(isinstance(cse_expr_as_tuple, CseExpr))
+    False
+
+    Parameters
+    ==========
```

Comment:
I think our standard order is Parameters before examples.

Reply:
The [docstrings style guide](https://docs.sympy.org/latest/contributing/docstring.html#sections) states examples 3rd and parameters 4th, which is why I've done it in this order.

Reply:
Well that's unfortunate :)

Reply:
It looks like all of the other docstrings in the module follow a different order of sections to that specified in the style guide. They do indeed have the parameters and returns sections before the examples sections. What do people advise here? Follow the style guide? Follow the module? Submit a following PR to update all docstrings in this module to match the style guide?

Reply:
Just follow the style guide. You can see that it has confused others too: https://github.com/sympy/sympy/issues/19591

I am sure we debated the order with the style guide went in. My opinion would be to follow the same order as the numpydoc guide (with parameters much closer to the function signature), but I think it was put in this order because many sympy docstrings don't have a parameters section and you have to see how to use it solely from the examples.